crypto/src/merkle_tree/proof.rs

```diff
@@ -11,25 +11,28 @@ use lambdaworks_math::{
 /// `i`-th element of `merkle_path` is the sibling node in the `n - 1 - i`-th check
 /// when verifying.
 #[derive(Debug, Clone)]
-pub struct Proof<F: IsField, H: IsCryptoHash<F>> {
+pub struct Proof<F: IsField> {
     pub merkle_path: Vec<FieldElement<F>>,
-    pub hasher: H,
 }
 
-impl<F: IsField, H: IsCryptoHash<F>> Proof<F, H> {
+impl<F: IsField> Proof<F> {
     pub fn verify(
         &self,
         root_hash: &FieldElement<F>,
         mut index: usize,
         value: &FieldElement<F>,
-    ) -> bool {
-        let mut hashed_value = self.hasher.hash_one(value.clone());
+        hasher: &dyn IsCryptoHash<F>,
+    ) -> bool
+    where
+        FieldElement<F>: ByteConversion,
+    {
+        let mut hashed_value = hasher.hash_one(value.clone());
```

Comment:
I find it odd that the hashers need to take ownership of their arguments. Why is that?

Reply:
You're right. Taking ownership should be pointless. I'll make the change and see how it goes